### DISCUSSION TITLE
Building projects from git-repo sources managed by chezmoiexternal

### DISCUSSION THREAD

**Discussion by dslatkin** [twpayne/chezmoi#2398] [Q&A]
There are some packages I want to install from source [like irssi](https://github.com/irssi/irssi) in order to have the most up-to-date version. Would it make sense to use `.chezmoiexternals` to clone and keep the source repo up-to-date and use scripts to actually build and place the executables in my `$PATH`?

**Comment by bradenhilton** [comment]
Seems like a reasonable approach. You could use a `run_onchange_after_` script with a few hash checks on the critical files similar to [Run a a script when the contents of another files changes](https://www.chezmoi.io/user-guide/use-scripts-to-perform-actions/#run-a-script-when-the-contents-of-another-file-changes) and https://github.com/twpayne/chezmoi/issues/2271#issuecomment-1220616225.

Have you tried something like this?
